api/api.proto

```diff
@@ -188,6 +188,28 @@ enum HardwareType {
   BareMetalMachine = 2;
 }
 
+// IOLimits per device IOLimits.
+message IOLimits {
+  // ReadBytesPerSecond read bytes per second.
+  uint64 read_bytes_per_second = 1;
+  // ReadIOPS read IOPS
+  uint64 read_iops = 2;
+  // WriteBytesPerSecond write bytes per second.
+  uint64 write_bytes_per_second = 3;
+  // WriteIOPS write IOPS
+  uint64 write_iops = 4;
+}
+
+// Qos per device Qos
+message Qos {
+  // Guaranteed IO limits.
+  IOLimits guaranteed = 1; 
+  // Limit throttles performance to these limits.
+  IOLimits limit = 2;
+  // Weight fair share weight in the range of [1..10]
+  uint32 weight = 3;
```

Comment:
Does this weight apply when QoS needs to be applied against two different volumes ?